core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -172,72 +138,210 @@ public void testDynamicBrokerConfigUpdateUsingZooKeeper() throws Exception {
-        encoderConfigs2.put(PasswordEncoderConfigs.PASSWORD_ENCODER_SECRET_CONFIG, "encoder-secret");
-        encoderConfigs2.put(PasswordEncoderConfigs.PASSWORD_ENCODER_CIPHER_ALGORITHM_CONFIG, "DES/CBC/PKCS5Padding");
-        encoderConfigs2.put(PasswordEncoderConfigs.PASSWORD_ENCODER_ITERATIONS_CONFIG, "1024");
-        encoderConfigs2.put(PasswordEncoderConfigs.PASSWORD_ENCODER_KEYFACTORY_ALGORITHM_CONFIG, "PBKDF2WithHmacSHA1");
-        encoderConfigs2.put(PasswordEncoderConfigs.PASSWORD_ENCODER_KEY_LENGTH_CONFIG, "64");
-        alterConfigWithZk(zkClient, configs2, Optional.of(brokerId), encoderConfigs2);
+        Map<String, String> configs2 = singletonMap("listener.name.internal.ssl.keystore.password", "secret2");
+        Map<String, String> encoderConfigs2 = generateEncodeConfig();
+        alterConfigWithZk(zkClient, Optional.of(brokerId), configs2, encoderConfigs2);
         Properties brokerConfigs2 = zkClient.getEntityConfigs("brokers", brokerId);
         String encodedPassword2 = brokerConfigs2.getProperty("listener.name.internal.ssl.keystore.password");
         assertEquals("secret2", ConfigCommand.createPasswordEncoder(JavaConverters.mapAsScalaMap(encoderConfigs)).decode(encodedPassword2).value());
         assertEquals("secret2", ConfigCommand.createPasswordEncoder(JavaConverters.mapAsScalaMap(encoderConfigs2)).decode(encodedPassword2).value());
 
         // Password config update at default cluster-level should fail
-        assertThrows(ConfigException.class, () -> alterConfigWithZk(zkClient, configs, Optional.empty(), encoderConfigs));
+        assertThrows(ConfigException.class, () -> alterConfigWithZk(zkClient, Optional.empty(), configs, encoderConfigs));
 
         // Dynamic config updates using ZK should fail if broker is running.
         registerBrokerInZk(zkClient, Integer.parseInt(brokerId));
-        assertThrows(IllegalArgumentException.class, () -> alterConfigWithZk(zkClient, Collections.singletonMap("message.max.size", "210000"), Optional.of(brokerId)));
-        assertThrows(IllegalArgumentException.class, () -> alterConfigWithZk(zkClient, Collections.singletonMap("message.max.size", "220000"), Optional.empty()));
+        assertThrows(IllegalArgumentException.class, () -> alterConfigWithZk(zkClient, Optional.of(brokerId), singletonMap("message.max.size", "210000")));
+        assertThrows(IllegalArgumentException.class, () -> alterConfigWithZk(zkClient, Optional.empty(), singletonMap("message.max.size", "220000")));
 
         // Dynamic config updates using ZK should for a different broker that is not running should succeed
-        alterAndVerifyConfig(zkClient, Collections.singletonMap("message.max.size", "230000"), Optional.of("2"));
+        alterAndVerifyConfig(zkClient, Optional.of("2"), singletonMap("message.max.size", "230000"));
+    }
+
+    private void alterAndVerifyConfig(KafkaZkClient zkClient, Optional<String> brokerId, Map<String, String>... configs) {
```

Comment:
We should avoid using variages for collection types. The callers should merge those collections into single one.